### ISSUE TITLE
[no-invalid-position-at-import-rule] Should allow `@layer` to come before `@import`

### ISSUE THREAD

**Issue by IanVS** [stylelint/stylelint#6085]
### What steps are needed to reproduce the bug?

```css
/* establish a layer order up-front, from lowest to highest priority */
@layer reset, defaults, theme, patterns, components, utilities, overrides;

@import url('colors.css') layer(theme);
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
	"no-invalid-position-at-import-rule": true,
}
```

### How did you run Stylelint?

stylelint 'src/**/*.css'

### Which version of Stylelint are you using?

`14.8.2`

### What did you expect to happen?

This is valid syntax according to the spec: https://www.w3.org/TR/css-cascade-5/#layer-empty

### What actually happened?

```
 4:1  ✖  Unexpected invalid position @import rule  no-invalid-position-at-import-rule
```

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

The rule should be updated to allow `@layer` to come before `@import`, and the rule's docs should be updated as well to reflect this.

**Comment by alexander-akait** [comment]
@g-plane We should track it for https://github.com/swc-project/swc/blob/main/crates/swc_css_lints/src/rules/no_invalid_position_at_import_rule.rs

**Comment by ybiquitous** [comment]
@IanVS Thanks for reporting the issue and using the template.

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

It should only be necessary to fix the `if` logic:

https://github.com/stylelint/stylelint/blob/479c99d394d9cae270d66a73d0c77ea74a3076cc/lib/rules/no-invalid-position-at-import-rule/index.js#L46-L53
